build-tools/timing/timing.targets

```diff
@@ -0,0 +1,152 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="timing.projitems" />
+  <Import Project="..\scripts\TestApks.targets" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.Git" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.GitCommitHash" />
+  <UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\xa-prep-tasks.dll" TaskName="Xamarin.Android.BuildTools.PrepTasks.ProcessMSBuildTiming" />
+  <Target Name="LogcatTiming">
+    <Exec Command="&quot;$(AdbToolPath)\$(AdbToolExe)&quot; $(_AdbTarget) $(AdbOptions) logcat -v threadtime -d > tmp-logcat-timing.log" />
+    <ProcessLogcatTiming
+        InputFilename="tmp-logcat-timing.log"
+        ApplicationPackageName="$(Package)"
+        PID="$(PID)"
+        ResultsFilename="tmp-timing-results.csv"
+        DefinitionsFilename="..\scripts\TimingDefinitions.txt"
+        AddResults="true"
+        Activity="$(Activity)" />
+  </Target>
+  <PropertyGroup>
+    <_MSBuildTimingDependsOn>
+      MSBuildPrep;
+      FreshBuild;
+      FreshInstall;
+      SecondBuild;
+      SecondInstall;
+      TouchCSharpBuild;
+      TouchCSharpInstall;
+      TouchAndroidResourceBuild;
+      TouchAndroidResourceInstall;
+    </_MSBuildTimingDependsOn>
+  </PropertyGroup>
+  <Target Name="MSBuildTiming" DependsOnTargets="$(_MSBuildTimingDependsOn)">
+    <ItemGroup>
+      <_TimingResults Include="$(_OutputDir)Timing_*.xml" />
+    </ItemGroup>
+    <ProcessMSBuildTiming
+        InputFiles="@(_TimingResults)"
+        ResultsFilename="$(_TopDir)TestResult-MSBuild-times.csv"
+        LabelSuffix="-$(Configuration)"
+    />
+  </Target>
+  <Target Name="MSBuildPrep" DependsOnTargets="ResolveReferences;AcquireAndroidTarget">
+    <PropertyGroup>
+      <_OutputDir>$(_TopDir)bin\Test$(Configuration)\</_OutputDir>
+      <_XABuild>$(_TopDir)bin\Debug\bin\xabuild</_XABuild>
+      <_TimingLogger>Xamarin.Android.Timing.TimingLogger,$(_OutputDir)Xamarin.Android.Timing.dll</_TimingLogger>
+    </PropertyGroup>
+    <ItemGroup>
+      <_FilesToDelete Include="$(_OutputDir)Timing_*" />
+    </ItemGroup>
+    <Delete Files="@(_FilesToDelete)"/>
+    <GitCommitHash
```

Comment:
Why not use the `GetXAVersionInfo` target from `build-tools/scripts/XAVersionInfo.targets`, which populates `$(XAVersionHash)`?